modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/AdapterRegistry.scala

```diff
@@ -34,29 +34,46 @@ import com.snowplowanalytics.snowplow.enrich.common.utils.HttpClient
  * The AdapterRegistry lets us convert a CollectorPayload into one or more RawEvents, using a given
  * adapter.
  */
-class AdapterRegistry(remoteAdapters: Map[(String, String), RemoteAdapter] = Map.empty) {
-
+class AdapterRegistry(
+  remoteAdapters: Map[(String, String), RemoteAdapter] = Map.empty,
+  callrailConfig: CallrailConfig,
+  cloudfrontAccessLogConfig: CloudfrontAccessLogConfig,
+  googleAnalyticsConfig: GoogleAnalyticsConfig,
+  hubspotConfig: HubspotConfig,
+  mailchimpConfig: MailchimpConfig,
+  mailgunConfig: MailgunConfig,
+  mandrillConfig: MandrillConfig,
+  marketoConfig: MarketoConfig,
+  olarkConfig: OlarkConfig,
+  pagerdutyConfig: PagerdutyConfig,
+  pingdomConfig: PingdomConfig,
+  sendgridConfig: SendgridConfig,
+  statusGatorConfig: StatusGatorConfig,
+  unbounceConfig: UnbounceConfig,
+  urbanAirshipConfig: UrbanAirshipConfig,
+  veroConfig: VeroConfig
+) {
   val adapters: Map[(String, String), Adapter] = Map(
     (Vendor.Snowplow, "tp1") -> Tp1Adapter,
     (Vendor.Snowplow, "tp2") -> Tp2Adapter,
     (Vendor.Redirect, "tp2") -> RedirectAdapter,
     (Vendor.Iglu, "v1") -> IgluAdapter,
-    (Vendor.Callrail, "v1") -> CallrailAdapter,
-    (Vendor.Cloudfront, "wd_access_log") -> CloudfrontAccessLogAdapter,
-    (Vendor.Mailchimp, "v1") -> MailchimpAdapter,
-    (Vendor.Mailgun, "v1") -> MailgunAdapter,
-    (Vendor.GoogleAnalytics, "v1") -> GoogleAnalyticsAdapter,
-    (Vendor.Mandrill, "v1") -> MandrillAdapter,
-    (Vendor.Olark, "v1") -> OlarkAdapter,
-    (Vendor.Pagerduty, "v1") -> PagerdutyAdapter,
-    (Vendor.Pingdom, "v1") -> PingdomAdapter,
-    (Vendor.Sendgrid, "v3") -> SendgridAdapter,
-    (Vendor.StatusGator, "v1") -> StatusGatorAdapter,
-    (Vendor.Unbounce, "v1") -> UnbounceAdapter,
-    (Vendor.UrbanAirship, "v1") -> UrbanAirshipAdapter,
-    (Vendor.Marketo, "v1") -> MarketoAdapter,
-    (Vendor.Vero, "v1") -> VeroAdapter,
-    (Vendor.HubSpot, "v1") -> HubSpotAdapter
+    (Vendor.Callrail, "v1") -> new CallrailAdapter(callrailConfig),
```

Comment:
Using `case class` instead of `class` will remove the need to use `new` and will be closer to the `object` that was there before

Reply:
Done!